controllers/componentversion_controller.go

```diff
@@ -152,57 +134,61 @@ func (r *ComponentVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		conditions.MarkTrue(obj,
 			meta.ReadyCondition,
 			meta.SucceededReason,
-			fmt.Sprintf("Applied version: %s", version))
+			"Applied version: %s",
+			version)
 
 		return ctrl.Result{
 			RequeueAfter: obj.GetRequeueAfter(),
 		}, nil
 	}
 
+	rreconcile.ProgressiveStatus(false, obj, meta.ProgressingReason, "updating component to new version: %s: %s", obj.Spec.Component, version)
+
 	ok, err := r.OCMClient.VerifyComponent(ctx, octx, obj, version)
 	if err != nil {
-		msg := fmt.Sprintf("failed to verify %s with constraint %s with error: %s", obj.Spec.Component, obj.Spec.Version.Semver, err)
-		conditions.Delete(obj, meta.ReconcilingCondition)
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.VerificationFailedReason, msg)
-		event.New(r.EventRecorder, obj, eventv1.EventSeverityError, fmt.Sprintf("%s, retrying in %s", err.Error(), obj.GetRequeueAfter()), nil)
+		MarkAsFailed(
+			r.EventRecorder,
+			obj,
+			v1alpha1.VerificationFailedReason,
+			fmt.Sprintf("failed to verify %s with constraint %s with error: %s", obj.Spec.Component, obj.Spec.Version.Semver, err),
+		)
 
 		return ctrl.Result{
 			RequeueAfter: obj.GetRequeueAfter(),
 		}, nil
 	}
 
 	if !ok {
-		msg := "attempted to verify component, but the digest didn't match"
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.VerificationFailedReason, msg)
-		event.New(r.EventRecorder, obj, eventv1.EventSeverityError, fmt.Sprintf("%s, retrying in %s", msg, obj.GetRequeueAfter()), nil)
+		MarkAsFailed(
+			r.EventRecorder,
+			obj,
+			v1alpha1.VerificationFailedReason,
+			"attempted to verify component, but the digest didn't match",
+		)
 
 		return ctrl.Result{
 			RequeueAfter: obj.GetRequeueAfter(),
 		}, nil
 	}
 
-	// update the result for the defer call to have the latest information
-	rresult, err := r.reconcile(ctx, octx, obj, version)
-	if err != nil {
-		event.New(r.EventRecorder, obj, eventv1.EventSeverityError, fmt.Sprintf("Reconciliation failed: %s, retrying in %s", err.Error(), obj.GetRequeueAfter()), nil)
-	}
-
-	return rresult, err
+	return r.reconcile(ctx, octx, obj, version)
 }
 
 func (r *ComponentVersionReconciler) reconcile(ctx context.Context, octx ocm.Context, obj *v1alpha1.ComponentVersion, version string) (ctrl.Result, error) {
```

Comment:
maybe `reconcileLatestVersion`?

Reply:
Technically it's not always the latest version. :D Or not even a version if it ends up not reconciling. :)